src/Concordium/Client/Runner.hs

```diff
@@ -2255,15 +2255,29 @@ processBakerCmd action baseCfgDir verbose backend =
       withClient backend $ do
          let senderAddr = naAddr . esdAddress . tcEncryptedSigningData $ txCfg
          AccountInfoResult{..} <- getAccountInfoOrDie senderAddr
+         energyrate <- getNrgGtuRate
+         let gtuTransactionPrice = Types.computeCost energyrate (tcEnergy txCfg) 
          case airBaker of
            Just AccountInfoBakerResult{..} -> logFatal [[i|Account is already a baker with ID #{aibiIdentity abirAccountBakerInfo}.|]]
            Nothing -> do
-             -- TODO: this should also take into account the estimated cost for this transaction
-             if airAmount < initialStake
+             if (airAmount - gtuTransactionPrice) < initialStake
              then
-               logFatal [[i|Account balance (#{showGtu airAmount}) is lower than the amount requested to be staked (#{showGtu initialStake}).|]]
+               logFatal [[i|Account balance (#{showGtu airAmount}) minus the cost of the transaction (#{showGtu gtuTransactionPrice}) is lower than the amount requested to be staked (#{showGtu initialStake}).|]]
              else do
                sendAndMaybeOutputCredentials bakerKeys accountKeysFile outputFile txCfg pl intOpts
+        where
+          -- General function to fetch NrgGtu rate
+          -- This functionality is going to be more generally used in an upcoming branch adding GTU prices to NRG printouts across the client.
+          -- Invoking it directly here to avoid extra work on compatability/refactoring in the future
+          getNrgGtuRate :: ClientMonad IO Types.EnergyRate
+          getNrgGtuRate = do 
+            blockSummary <- getFromJson =<< withBestBlockHash Nothing getBlockSummary
+            case blockSummary of
+              Nothing -> do
+                logError ["Failed to retrieve NRG-GTU rate from best block hash, unable to estimate GTU cost"]
+                return 0
```

Comment:
I think probably it is better if this returns a Maybe instead of 0. This can then be handled by the consumer.

Alternatively the error message should say something to the effect "Using default exchange rate ..."

Reply:
I added some more information to the error message in the latest commit, specifying that the "default" behaviour is that it falls back to print all NRG-derived GTU values as 0. 
This, (now that it's more clearly communicated,) feels like the easiest/safest default behaviour, given that this function should for now exclusively be used for computing GTU values for informational prints.

If 0 isn't ideal, is there a different default value you would recommend?

Alternatives could be erroring out, or asking the user to rerun the command (assuming the edgecase of best block being dropped by the node due to finalization/branch switches in between calls.)

Maybe feels like overkill, but could be an option if we want to use it for more "mission critical" functionality later.

Reply:
I think this is Ok, but I would document this caveat about the function, especially if this is going to be a top-level function in the future.